nntrainer/dataset/random_data_producers.cpp

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 Jihoon Lee <jhoon.it.lee@samsung.com>
+ *
+ * @file   random_data_producers.cpp
+ * @date   09 July 2021
+ * @brief  This file contains various random data producers
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Jihoon Lee <jhoon.it.lee@samsung.com>
+ * @bug    No known bugs except for NYI items
+ *
+ */
+#include <random_data_producers.h>
+
+#include <base_properties.h>
+#include <node_exporter.h>
+#include <util_func.h>
+
+namespace nntrainer {
+
+/**
+ * @brief Props containing min value
+ *
+ */
+class PropsMin : public Property<float> {
+public:
+  /**
+   * @brief Construct a new props min object with a default value
+   *
+   * @param value default value
+   */
+  PropsMin(float value = 0.0f) : nntrainer::Property<float>(value) {}
+  static constexpr const char *key = "min"; /**< unique key to access */
+  using prop_tag = float_prop_tag;          /**< property type */
+};
+
+/**
+ * @brief Props containing max value
+ *
+ */
+class PropsMax : public Property<float> {
+public:
+  /**
+   * @brief Construct a new props max object with a default value
+   *
+   * @param value default value
+   */
+  PropsMax(float value = 1.0f) : nntrainer::Property<float>(value) {}
+  static constexpr const char *key = "max"; /**< unique key to access */
```

Comment:
does it requre comparison with PropsMin when it set?? If it is to be property, then how we check when it is assigned as we planed... Just check type?